### PR TITLE
 Add Brackets editor Code to known editors for coala

### LINKED ISSUE #5829: Add Brackets editor Code to known editors for coala

### PR DESCRIPTION
**Pull request by XLAR-8**
Added bracket to the list of known GUI Editors.

Guidance Required:
In order to open a file , the command goes:

Open a file in the current folder,
without switching projects:
brackets index.html

Open two files from a specific path,
without switching projects:
brackets ~/Sites/index.html ~/Sites/index.css

Open the current folder as a project in Brackets:
brackets .

Open a specific folder as a project in Brackets:
brackets ~/Sites/mysite

Brackets can be launched from the command line.
Presently it does not accept any other arguments.
Any updates required will be implemented.
Closes #5829

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by amalrajan** [comment]
Shorten the commit title (max: 50); move any descriptive information to the commit message box.

**Comment by XLAR-8** [comment]
@amalrajan @utkarsh2102  thank you very much for the solution .

**Comment by XLAR-8** [comment]
@amalrajan @utkarsh2102  what are these errors  :
https://circleci.com/gh/coala/coala/19699?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
https://travis-ci.org/coala/coala/jobs/443698510

**Comment by XLAR-8** [comment]
and the issue is still pending review , what should i do create new pull request or wait ?
@jayvdb  please guide

File: coalib/results/result_actions/OpenEditorAction.py [hunk 1]
```diff
@@ -96,6 +96,10 @@
         'args': '--new-window',
         'gui': True
     },
+    'brackets': {
```
Comment (jayvdb) [inline coalib/results/result_actions/OpenEditorAction.py]: alpha order please

Comment (XLAR-8) [reply coalib/results/result_actions/OpenEditorAction.py]: @jayvdb I checked, brackets does not support jumping to a line number.

**Comment by XLAR-8** [comment]
@jayvdb , I made all necessary changes but still errors not seem to be reducing .
shoul i delete this pull and start new one .
please guide.

**Comment by jayvdb** [comment]
Never close a Pull Request. You need to learn how to fix them.

First step is to squash your 5 commits, which you can see at https://github.com/coala/coala/pull/5831/commits ,  into a single commit.

The easiest way to do that, is to delete the top 4 commits, erasing your incorrect extra commits.

To do that, you find the identifier of the first commit in your series, which was https://github.com/coala/coala/pull/5831/commits/68b6cf080b468d661ef7d5f325367690355079f5 , and reset to that position.

`git reset --hard 68b6cf080b`

Then edit the file again, but make sure you amend your first commit, instead of creating new commits.

`git add coalib/results/result_actions/OpenEditorAction.py`
`git commit --amend`

**Comment by shivank1234** [comment]
Squash 5 commits into 1 locally and then push them to your fork,you can look onto the following link for it
https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit
Follow the guidelines given in https://api.coala.io/en/latest/Developers/Writing_Good_Commits.html to write the commit message properly and you are good to go..

File: coalib/results/result_actions/OpenEditorAction.py [hunk 2]
```diff
@@ -63,6 +63,10 @@
         'args': '--wait',
         'gui': True
     },
+    'brackets': {
+        'file_arg_template': '{filename}',
+        'gui': True
```
Comment (jayvdb) [inline coalib/results/result_actions/OpenEditorAction.py]: add trailing comma

**Pull request closed without merging**